homeassistant/components/fyta/__init__.py

```diff
@@ -47,3 +54,38 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         hass.data[DOMAIN].pop(entry.entry_id)
 
     return unload_ok
+
+
+async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Migrate old entry."""
+    _LOGGER.debug("Migrating from version %s", config_entry.version)
+
+    if config_entry.version > 1:
+        # This means the user has downgraded from a future version
+        return False
+
+    if config_entry.version == 1:
+        new = {**config_entry.data}
+        if config_entry.minor_version < 1:
+            fyta = FytaConnector(
+                config_entry.data[CONF_USERNAME], config_entry.data[CONF_PASSWORD]
+            )
+            credentials: dict[str, str | datetime] = fyta.login()
+
+            if isinstance(credentials["expiration"], datetime):
+                credentials["expiration"] = credentials["expiration"].isoformat()
```

Comment:
Let's type credentials as str, Any.

I think it's safe to assume expiration is typed as date time. As improvement point you could look into creating it an object in the fyta Library